src/main/java/com/aws/greengrass/smbridge/clients/MQTTClient.java

```diff
@@ -10,52 +10,53 @@
 import com.aws.greengrass.logging.api.Logger;
 import com.aws.greengrass.logging.impl.LogManager;
 import com.aws.greengrass.smbridge.SMBridge;
-import com.aws.greengrass.smbridge.Message;
+import com.aws.greengrass.smbridge.MQTTMessage;
 import com.aws.greengrass.smbridge.auth.MQTTClientKeyStore;
 import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.Utils;
 import lombok.AccessLevel;
 import lombok.Getter;
-import org.eclipse.paho.client.mqttv3.IMqttDeliveryToken;
-import org.eclipse.paho.client.mqttv3.MqttCallback;
-import org.eclipse.paho.client.mqttv3.MqttClient;
-import org.eclipse.paho.client.mqttv3.MqttClientPersistence;
-import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
-import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.*;
 import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence;
 
 import java.security.KeyStoreException;
 import java.util.HashSet;
 import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
 import java.util.function.Consumer;
 import javax.inject.Inject;
 import javax.net.ssl.SSLSocketFactory;
 
 public class MQTTClient {
     private static final Logger LOGGER = LogManager.getLogger(MQTTClient.class);
     private static final String DEFAULT_BROKER_URI = "ssl://localhost:8883";
+    private static final String DEFAULT_CLIENT_ID = "sm-bridge-" + Utils.generateRandomString(11);
```

Comment:
Why are you using a random string. Also using a random  Coerce.toString(topics.findOrDefault(DEFAULT_CLIENT_ID,
                KernelConfigResolver.CONFIGURATION_CONFIG_KEY, CLIENT_ID_KEY)); will not work properly.

Reply:
This is copied from the MQTT Bridge. I believe it is random so that every component running on a different device has a unique Client ID. I'll ping you about the coercion.